app/src/renderer/vuex/modules/wallet.js

```diff
@@ -86,42 +83,23 @@ export default ({ node }) => {
         state.error = error
       }
     },
-    async loadDenoms({ commit, state }, maxIterations = 10) {
-      // read genesis.json to get default denoms
-
-      // wait for genesis.json to exist
-      let genesisPath = join(root, `genesis.json`)
-
-      // wait for the genesis and load it
-      // at some point give up and throw an error
-      while (maxIterations) {
-        try {
-          await fs.pathExists(genesisPath)
-          break
-        } catch (error) {
-          console.log(`waiting for genesis`, error, genesisPath)
-          maxIterations--
-          await sleep(500)
-        }
-      }
-      if (maxIterations === 0) {
-        const error = new Error(`Couldn't load genesis at path ${genesisPath}`)
-        Sentry.captureException(error)
-        state.error = error
-        return
-      }
+    async loadDenoms({ commit, state }) {
+      try {
+        const { genesis } = await network()
```

Comment:
Only place where the genesis is needed. Maybe query from stargate?

Reply:
that would be awesome! It will allow us to know just a stargate address to start